examples/go/stream_benchmarker/benchmark/main.go

```diff
@@ -0,0 +1,412 @@
+			metricsData.mostRecentTimeLastByteReceived = metric.timeLastByteReceived
+		}
+
+		if previousFirstByteTime.After(firstByteTime) {
+			outOfOrderData++
+		}
+
+		previousFirstByteTime = firstByteTime
+	}
+
+	if metricsCount > 0 {
+		averageFirstByteLatency = totalFirstByteLatency / time.Duration(metricsCount)
+		averageLastByteLatency = totalLastByteLatency / time.Duration(metricsCount)
+		averageDataSize = float64(totalDataSize) / float64(metricsCount)
+	}
+
+	dataRateInMbps := (float64(totalDataSize) / (now.Sub(metricsData.initialTime).Seconds())) * 8 / 1024 / 1024
+
+	planID = metricsData.planID
+
+	outputDetails := outputDetails{
+		now:                            metricsData.timeOfReporting,
+		mostRecentTimeLastByteReceived: metricsData.mostRecentTimeLastByteReceived,
+		averageFirstByteLatency:        averageFirstByteLatency,
+		averageLastByteLatency:         averageLastByteLatency,
+		totalDataSize:                  totalDataSize,
+		metricsCount:                   metricsCount,
+		averageDataSize:                averageDataSize,
+		dataRateInMbps:                 dataRateInMbps,
+		outOfOrderData:                 outOfOrderData,
+		planID:                         planID,
+	}
+
+	return outputDetails
+}
+
+func printHeader(output io.Writer, planID string) {
+	fmt.Fprintf(output, "Pass %v:\n%10v%25v%25v%20v%18v%18v%15v%16v%11v%20v",
+		planID,
+		"PlanID",
+		"DATE",
+		"Most recent",
+		"First byte latency",
+		"Last byte latency",
+		"Total bytes",
+		"Num messages",
+		"Avg bytes",
+		"Mbps",
+		"Out of order\n")
+}
+
+func printPassDetails(output io.Writer, outputDetails outputDetails) {
+	fmt.Fprintf(output, "%10v%25v%25v%20.2f%18.2f%18v%15v%16.2f%11.2f%20v\n",
+		outputDetails.planID,
+		outputDetails.now.Format(dateTimeFormat),
+		outputDetails.mostRecentTimeLastByteReceived.Format(dateTimeFormat),
+		outputDetails.averageFirstByteLatency.Seconds(),
+		outputDetails.averageLastByteLatency.Seconds(),
+		outputDetails.totalDataSize,
+		outputDetails.metricsCount,
+		outputDetails.averageDataSize,
+		outputDetails.dataRateInMbps,
+		outputDetails.outOfOrderData)
+}
+
+func printPassSummary(output io.Writer, ps passSummary) {
+	fmt.Fprintf(output, "\nPass Summary:\n%10v%25v%25v%20v%20v%20v%15v%15v\n",
+		"PlanID",
+		"First byte time",
+		"Last byte time",
+		"First byte batency",
+		"Last byte latency",
+		"Total bytes",
+		"Average Mbps",
+		"Out of order")
+
+	averageFirstByteLatency := ps.totalAverageFirstByteLatency.Seconds() / ps.numIntervals
+	averageLastByteLatency := ps.totalAverageLastByteLatency.Seconds() / ps.numIntervals
+
+	fmt.Fprintf(output, "%10v%25v%25v%20.2f%20.2f%20v%15.2f%15v\n\n",
+		ps.initialPlanID,
+		ps.firstByteTime.Format(dateTimeFormat),
+		ps.lastByteTime.Format(dateTimeFormat),
+		averageFirstByteLatency,
+		averageLastByteLatency,
+		ps.totalBytes,
+		ps.totalMbps/ps.numIntervals,
+		ps.totalOutOfOrderData)
+}
+
+func (sessionSummary *sessionSummary) printSessionSummary(output io.Writer) {
```

Comment:
Since this is on `sessionSummary`, you can just call it `print(output io.Writer)`.